[SPARK-14988][PYTHON] SparkSession API follow-ups

andrewor14 opened the pull request:
## What changes were proposed in this pull request?

Addresses comments in #12765.
## How was this patch tested?

Python tests.

Commit:
registerDataFrameAsTable -> registerTable

Commit:
getOption -> get with default

Commit:
Move conf tests

Commit:
Move catalog tests

Commit:
Fix style

andrewor14 commented:
@davies

SparkQA commented:
**[Test build #57355 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/57355/consoleFull)** for PR 12784 at commit [`ee606dd`](https://github.com/apache/spark/commit/ee606dd9425cb5ec28273477dd4b560b4cb6bd3e).

Commit:
Merge branch 'master' of github.com:apache/spark into python-followup

Commit:
Correct conf default behavior

When a conf has a default, then the user provided default was
never read. Instead, the default default should be overridden
by the user provided default.

SparkQA commented:
**[Test build #57360 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/57360/consoleFull)** for PR 12784 at commit [`a8b0732`](https://github.com/apache/spark/commit/a8b0732392599e3af86137b0d207aa16eb27f5da).

SparkQA commented:
**[Test build #57355 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/57355/consoleFull)** for PR 12784 at commit [`ee606dd`](https://github.com/apache/spark/commit/ee606dd9425cb5ec28273477dd4b560b4cb6bd3e).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/57355/
Test FAILed.

davies commented on python/pyspark/sql/conf.py:
```diff
@@ -33,62 +33,24 @@ def __init__(self, jconf):
+            return self._jconf.get(key)
         else:
-            return None
+            return self._jconf.get(key, default)
```
Could you check the type for key and default, or you will get an weird py4j error?

davies commented on python/pyspark/sql/tests.py:
```diff
@@ -1394,6 +1395,200 @@ def test_toDF_with_schema_string(self):
+        self.assertEquals(sorted(databases), ["default", "some_db"])
+
+    def test_list_tables(self):
+        from catalog import Table
```
It's better to use full path
from pyspark.sql.catalog import Table

Commit:
Address more comments

SparkQA commented:
**[Test build #57365 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/57365/consoleFull)** for PR 12784 at commit [`768fd81`](https://github.com/apache/spark/commit/768fd816a3c7b0c49eb5eaf43cb322d17e94d9a2).

SparkQA commented:
**[Test build #57368 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/57368/consoleFull)** for PR 12784 at commit [`059bfdd`](https://github.com/apache/spark/commit/059bfdd73d10941bc1d314735d78bafc99149d9b).

SparkQA commented:
**[Test build #2940 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2940/consoleFull)** for PR 12784 at commit [`059bfdd`](https://github.com/apache/spark/commit/059bfdd73d10941bc1d314735d78bafc99149d9b).

SparkQA commented:
**[Test build #2941 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2941/consoleFull)** for PR 12784 at commit [`059bfdd`](https://github.com/apache/spark/commit/059bfdd73d10941bc1d314735d78bafc99149d9b).

SparkQA commented:
**[Test build #2942 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2942/consoleFull)** for PR 12784 at commit [`059bfdd`](https://github.com/apache/spark/commit/059bfdd73d10941bc1d314735d78bafc99149d9b).

SparkQA commented:
**[Test build #57360 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/57360/consoleFull)** for PR 12784 at commit [`a8b0732`](https://github.com/apache/spark/commit/a8b0732392599e3af86137b0d207aa16eb27f5da).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/57360/
Test FAILed.

SparkQA commented:
**[Test build #57365 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/57365/consoleFull)** for PR 12784 at commit [`768fd81`](https://github.com/apache/spark/commit/768fd816a3c7b0c49eb5eaf43cb322d17e94d9a2).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/57365/
Test PASSed.

SparkQA commented:
**[Test build #2942 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2942/consoleFull)** for PR 12784 at commit [`059bfdd`](https://github.com/apache/spark/commit/059bfdd73d10941bc1d314735d78bafc99149d9b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

SparkQA commented:
**[Test build #2941 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2941/consoleFull)** for PR 12784 at commit [`059bfdd`](https://github.com/apache/spark/commit/059bfdd73d10941bc1d314735d78bafc99149d9b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

SparkQA commented:
**[Test build #57368 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/57368/consoleFull)** for PR 12784 at commit [`059bfdd`](https://github.com/apache/spark/commit/059bfdd73d10941bc1d314735d78bafc99149d9b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/57368/
Test PASSed.

SparkQA commented:
**[Test build #2940 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/2940/consoleFull)** for PR 12784 at commit [`059bfdd`](https://github.com/apache/spark/commit/059bfdd73d10941bc1d314735d78bafc99149d9b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

andrewor14 commented:
Merging into master. Thanks @davies.
